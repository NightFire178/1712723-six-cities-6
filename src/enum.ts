export enum placeFavoriteButtonEnum {
  card = `card`,
  property = `property`
}

export enum cardPlaceEnum {
  cities = `cities`,
  near = `near`,
  favorites = `favorites`
}
