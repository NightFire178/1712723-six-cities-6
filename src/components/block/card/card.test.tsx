import React from 'react'
import {render} from '@testing-library/react'
import Card from './card'
import mockOneHotel from "../../../mock/mock-hotels";
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'
import {cardPlaceEnum} from "../../../enum";

//eslint-disable-next-line
jest.mock('../favorite-button/favorite-button', () => () => (<><div>favorite button component</div></>))


test(`Card component block test`, () => {
  const hotel = mockOneHotel
  const history = createMemoryHistory()
  const city = render(
    <Router history={history}>
      <Card objCard={hotel} cardPlace={cardPlaceEnum.cities}/>
    </Router>
    )
  expect(city).toMatchSnapshot()
  const favorites = render(
    <Router history={history}>
      <Card objCard={hotel} cardPlace={cardPlaceEnum.favorites}/>
    </Router>
  )
  expect(favorites).toMatchSnapshot()
  const near = render(
    <Router history={history}>
      <Card objCard={hotel} cardPlace={cardPlaceEnum.near}/>
    </Router>
  )
  expect(near).toMatchSnapshot()
})
