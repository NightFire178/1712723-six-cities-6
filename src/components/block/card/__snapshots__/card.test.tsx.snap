// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component block test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="cities__place-card place-card"
      >
        <div
          class="place-card__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          class="cities__image-wrapper place-card__image-wrapper"
        >
          <a
            href="/offer/1"
          >
            <img
              alt="Place image"
              class="place-card__image"
              height="200"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg"
              width="260"
            />
          </a>
        </div>
        <div
          class="place-card__info"
        >
          <div
            class="place-card__price-wrapper"
          >
            <div
              class="place-card__price"
            >
              <b
                class="place-card__price-value"
              >
                €
                169
              </b>
              <span
                class="place-card__price-text"
              >
                / night
              </span>
            </div>
            <div>
              favorite button component
            </div>
          </div>
          <div
            class="place-card__rating rating"
          >
            <div
              class="place-card__stars rating__stars"
            >
              <span
                style="width: 60%;"
              />
              <span
                class="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            class="place-card__name"
          >
            <a
              href="/offer/1"
            >
              The house among olive 
            </a>
          </h2>
          <p
            class="place-card__type"
          >
            room
          </p>
        </div>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="cities__place-card place-card"
    >
      <div
        class="place-card__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        class="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="/offer/1"
        >
          <img
            alt="Place image"
            class="place-card__image"
            height="200"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg"
            width="260"
          />
        </a>
      </div>
      <div
        class="place-card__info"
      >
        <div
          class="place-card__price-wrapper"
        >
          <div
            class="place-card__price"
          >
            <b
              class="place-card__price-value"
            >
              €
              169
            </b>
            <span
              class="place-card__price-text"
            >
              / night
            </span>
          </div>
          <div>
            favorite button component
          </div>
        </div>
        <div
          class="place-card__rating rating"
        >
          <div
            class="place-card__stars rating__stars"
          >
            <span
              style="width: 60%;"
            />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          class="place-card__name"
        >
          <a
            href="/offer/1"
          >
            The house among olive 
          </a>
        </h2>
        <p
          class="place-card__type"
        >
          room
        </p>
      </div>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card component block test 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="cities__place-card place-card"
      >
        <div
          class="place-card__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          class="cities__image-wrapper place-card__image-wrapper"
        >
          <a
            href="/offer/1"
          >
            <img
              alt="Place image"
              class="place-card__image"
              height="200"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg"
              width="260"
            />
          </a>
        </div>
        <div
          class="place-card__info"
        >
          <div
            class="place-card__price-wrapper"
          >
            <div
              class="place-card__price"
            >
              <b
                class="place-card__price-value"
              >
                €
                169
              </b>
              <span
                class="place-card__price-text"
              >
                / night
              </span>
            </div>
            <div>
              favorite button component
            </div>
          </div>
          <div
            class="place-card__rating rating"
          >
            <div
              class="place-card__stars rating__stars"
            >
              <span
                style="width: 60%;"
              />
              <span
                class="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            class="place-card__name"
          >
            <a
              href="/offer/1"
            >
              The house among olive 
            </a>
          </h2>
          <p
            class="place-card__type"
          >
            room
          </p>
        </div>
      </article>
    </div>
    <div>
      <article
        class="favorites__card place-card"
      >
        <div
          class="place-card__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          class="favorites__image-wrapper place-card__image-wrapper"
        >
          <a
            href="/offer/1"
          >
            <img
              alt="Place image"
              class="place-card__image"
              height="110"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg"
              width="150"
            />
          </a>
        </div>
        <div
          class="favorites__card-info place-card__info"
        >
          <div
            class="place-card__price-wrapper"
          >
            <div
              class="place-card__price"
            >
              <b
                class="place-card__price-value"
              >
                €
                169
              </b>
              <span
                class="place-card__price-text"
              >
                / night
              </span>
            </div>
            <div>
              favorite button component
            </div>
          </div>
          <div
            class="place-card__rating rating"
          >
            <div
              class="place-card__stars rating__stars"
            >
              <span
                style="width: 60%;"
              />
              <span
                class="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            class="place-card__name"
          >
            <a
              href="/offer/1"
            >
              The house among olive 
            </a>
          </h2>
          <p
            class="place-card__type"
          >
            room
          </p>
        </div>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="favorites__card place-card"
    >
      <div
        class="place-card__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        class="favorites__image-wrapper place-card__image-wrapper"
      >
        <a
          href="/offer/1"
        >
          <img
            alt="Place image"
            class="place-card__image"
            height="110"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg"
            width="150"
          />
        </a>
      </div>
      <div
        class="favorites__card-info place-card__info"
      >
        <div
          class="place-card__price-wrapper"
        >
          <div
            class="place-card__price"
          >
            <b
              class="place-card__price-value"
            >
              €
              169
            </b>
            <span
              class="place-card__price-text"
            >
              / night
            </span>
          </div>
          <div>
            favorite button component
          </div>
        </div>
        <div
          class="place-card__rating rating"
        >
          <div
            class="place-card__stars rating__stars"
          >
            <span
              style="width: 60%;"
            />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          class="place-card__name"
        >
          <a
            href="/offer/1"
          >
            The house among olive 
          </a>
        </h2>
        <p
          class="place-card__type"
        >
          room
        </p>
      </div>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card component block test 3`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="cities__place-card place-card"
      >
        <div
          class="place-card__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          class="cities__image-wrapper place-card__image-wrapper"
        >
          <a
            href="/offer/1"
          >
            <img
              alt="Place image"
              class="place-card__image"
              height="200"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg"
              width="260"
            />
          </a>
        </div>
        <div
          class="place-card__info"
        >
          <div
            class="place-card__price-wrapper"
          >
            <div
              class="place-card__price"
            >
              <b
                class="place-card__price-value"
              >
                €
                169
              </b>
              <span
                class="place-card__price-text"
              >
                / night
              </span>
            </div>
            <div>
              favorite button component
            </div>
          </div>
          <div
            class="place-card__rating rating"
          >
            <div
              class="place-card__stars rating__stars"
            >
              <span
                style="width: 60%;"
              />
              <span
                class="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            class="place-card__name"
          >
            <a
              href="/offer/1"
            >
              The house among olive 
            </a>
          </h2>
          <p
            class="place-card__type"
          >
            room
          </p>
        </div>
      </article>
    </div>
    <div>
      <article
        class="favorites__card place-card"
      >
        <div
          class="place-card__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          class="favorites__image-wrapper place-card__image-wrapper"
        >
          <a
            href="/offer/1"
          >
            <img
              alt="Place image"
              class="place-card__image"
              height="110"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg"
              width="150"
            />
          </a>
        </div>
        <div
          class="favorites__card-info place-card__info"
        >
          <div
            class="place-card__price-wrapper"
          >
            <div
              class="place-card__price"
            >
              <b
                class="place-card__price-value"
              >
                €
                169
              </b>
              <span
                class="place-card__price-text"
              >
                / night
              </span>
            </div>
            <div>
              favorite button component
            </div>
          </div>
          <div
            class="place-card__rating rating"
          >
            <div
              class="place-card__stars rating__stars"
            >
              <span
                style="width: 60%;"
              />
              <span
                class="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            class="place-card__name"
          >
            <a
              href="/offer/1"
            >
              The house among olive 
            </a>
          </h2>
          <p
            class="place-card__type"
          >
            room
          </p>
        </div>
      </article>
    </div>
    <div>
      <article
        class="near-places__card place-card"
      >
        <div
          class="place-card__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          class="near-places__image-wrapper place-card__image-wrapper"
        >
          <a
            href="/offer/1"
          >
            <img
              alt="Place image"
              class="place-card__image"
              height="200"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg"
              width="260"
            />
          </a>
        </div>
        <div
          class="place-card__info"
        >
          <div
            class="place-card__price-wrapper"
          >
            <div
              class="place-card__price"
            >
              <b
                class="place-card__price-value"
              >
                €
                169
              </b>
              <span
                class="place-card__price-text"
              >
                / night
              </span>
            </div>
            <div>
              favorite button component
            </div>
          </div>
          <div
            class="place-card__rating rating"
          >
            <div
              class="place-card__stars rating__stars"
            >
              <span
                style="width: 60%;"
              />
              <span
                class="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            class="place-card__name"
          >
            <a
              href="/offer/1"
            >
              The house among olive 
            </a>
          </h2>
          <p
            class="place-card__type"
          >
            room
          </p>
        </div>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="near-places__card place-card"
    >
      <div
        class="place-card__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        class="near-places__image-wrapper place-card__image-wrapper"
      >
        <a
          href="/offer/1"
        >
          <img
            alt="Place image"
            class="place-card__image"
            height="200"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg"
            width="260"
          />
        </a>
      </div>
      <div
        class="place-card__info"
      >
        <div
          class="place-card__price-wrapper"
        >
          <div
            class="place-card__price"
          >
            <b
              class="place-card__price-value"
            >
              €
              169
            </b>
            <span
              class="place-card__price-text"
            >
              / night
            </span>
          </div>
          <div>
            favorite button component
          </div>
        </div>
        <div
          class="place-card__rating rating"
        >
          <div
            class="place-card__stars rating__stars"
          >
            <span
              style="width: 60%;"
            />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          class="place-card__name"
        >
          <a
            href="/offer/1"
          >
            The house among olive 
          </a>
        </h2>
        <p
          class="place-card__type"
        >
          room
        </p>
      </div>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
